pkg/volume/csi/csi_client.go

```diff
@@ -31,12 +31,42 @@ import (
 	"k8s.io/kubernetes/pkg/features"
 )
 
-type csiClient interface {
+type csiNodeInfoGetter interface {
```

Comment:
Do we have to create a new interface with this? Could `GetNodeInfoWithFallback` just be a new interface method for csiClient

Reply:
Our thinking for not adding `GetNodeInfoWithFallback` as a additional interface method on `csiClient` was that we want to be able to write unit tests for the new method that use fakes of the original public methods `NodeGetInfo` and `NodeGetId`. That allows us to make clear assertions about their use. Therefore it's easier to write the code and tests when the method under test (`getNodeInfoWithFallback`) is not on the same object (and therefore not on the same interface).

Given that thought process, we introduced the new interface with just the two methods `NodeGetInfo` and `NodeGetId` so the tests could be written with fewer needless stubbed methods.